### DISCUSSION TITLE
Motivation for SDL backend mouse wheel handling

### DISCUSSION THREAD

**Discussion by Grieverheart** [ocornut/imgui#6081] [New Users Build/Link/Run issues ONLY!]
I'm wondering what the motivation is for handling the mouse wheel in SDL like this:
```C++
        case SDL_MOUSEWHEEL:
        {
            float wheel_x = (event->wheel.x > 0) ? 1.0f : (event->wheel.x < 0) ? -1.0f : 0.0f;
            float wheel_y = (event->wheel.y > 0) ? 1.0f : (event->wheel.y < 0) ? -1.0f : 0.0f;
            io.AddMouseWheelEvent(wheel_x, wheel_y);
            return true;
        }
```
why are increments clipped to +-1?

**Comment by PathogenDavid** [comment]
Whatever motivation there was has probably been lost to time. It's changed shape a few times over the years, but the general logic has been the same since the SDL backend was originally introduced back in 2015:

https://github.com/ocornut/imgui/blob/2acdafe4f2cf844c6e21d77e83b605b7aa1aa75c/examples/opengl_sdl_example/imgui_impl_sdl.cpp#L220-L229

It might just be that it was inspired by the example code from the SDL documentation, which does the same thing: https://wiki.libsdl.org/SDL2/SDL_MouseWheelEvent

**Comment by ocornut** [comment]
> I'm wondering what the motivation is for handling the mouse wheel in SDL like this:

Knowing which values are you getting and on what system may be useful to know. Thanks!

**Reply by Grieverheart** [reply]
@ocornut Should I create an issue for this? I removed the clipping on my local copy and it works as expected.

**Reply by ocornut** [reply]
Without an answer to my question, I can't really tell if it's worth taking further action.

**Reply by Grieverheart** [reply]
Sorry, which question? 😅

**Reply by Grieverheart** [reply]
Ah, sorry, I didn't see a question mark so I thought it was a general statement, with which I agreed :D . Anyway, on MacOS mouse wheel values can be way above 1 or below -1.

**Reply by ocornut** [reply]
Could you log the values for a typical scroll wheel flick and share that log?

**Reply by Grieverheart** [reply]
[wheel.txt](https://github.com/ocornut/imgui/files/10459959/wheel.txt)
I recorded two flicks. A slow and a fast one.

**Reply by ocornut** [reply]
Thanks. Linking to our work on #3795

**Reply by ocornut** [reply]
Do you have smooth scrolling disabled?
Could you try other backends: GLFW, OSX and compare the values you are getting?
Thanks a lot!

**Comment by ocornut** [comment]
See https://github.com/ocornut/imgui/issues/4019#issuecomment-1411007319 for a sneak peak at some messy the "mouse wheeling" landscape. Although I don't REMEMBER why we clamped it is apparent this is needed until we can unf*ckify this mess.

**Reply by ocornut** [reply]
See https://github.com/ocornut/imgui/issues/4019#issuecomment-1412698383 for a full document about this
(TL;DR; can't clamp just yet but when we figure out what to do to replace the GLFW scale we will try to apply same logic to SDL backend)
